### PR TITLE
CMR-6451: Autocomplete auth filtering

### PR DESCRIPTION
**Pull request by mschmele**
Update autocomplete indexing and search to not show private suggestions to users not logged in

### CODE REVIEW

**Comment by eereiter** [comment]
Does this also cover the case when users are logged in, but they don't have permission to view a record?  Or does this case already work?

**Comment by mschmele** [comment]
> Does this also cover the case when users are logged in, but they don't have permission to view a record? Or does this case already work?

It doesn't, but this work is going to need to be reimplemented once the elasticsearch version upgrade is deployed. This is the easiest way to filter suggestions without affecting performance, which was a big concern

File: indexer-app/src/cmr/indexer/services/index_service.clj [hunk 1]
```diff
@@ -288,7 +300,7 @@
                                        (debug %)
                                        nil)))
                              (remove nil?))
-        humanized-fields (map #(humanizer/collection-humanizers-elastic context %) parsed-concepts)
+        humanized-fields  (map #(humanizer/collection-humanizers-for-suggestion-docs context %) parsed-concepts)
```
Comment (jaybarra) [inline indexer-app/src/cmr/indexer/services/index_service.clj:303]: nitpick: extra space added

**Comment by mreese84** [comment]
Just looked at this with Mark. I agree with Leo. This feature should look at ACLs and not AccessConstraints. @mschmele, I recommend closing this PR and opening a new one when you get it reworked.

**Pull request closed without merging**